core/frontend/src/RealityDataConnection.ts

```diff
@@ -0,0 +1,118 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+
+import { GuidString } from "@bentley/bentleyjs-core";
+import { RealityDataSourceKey, RealityDataSourceProps } from "@bentley/imodeljs-common";
+import { RealityData, RealityDataClient } from "@bentley/reality-data-client";
+import { AuthorizedFrontendRequestContext } from "./FrontendRequestContext";
+import { RealityDataSource, realityDataSourceKeyToString } from "./RealityDataSource";
+
+/**
+ * This interface provide methods used to access a reality data from ContextShare
+ * @alpha
+ */
+export interface IRealityDataConnection {
+  getRealityData(): RealityData | undefined;
+  getRealityDataType(): string | undefined;
+  getServiceUrl(iTwinId: GuidString | undefined): Promise<string | undefined>;
+  getSource(): RealityDataSource;
+}
+
+/**
+ * This class encapsulates access to a reality data from ContextShare
+ * There is a one to one relationship between a reality data and the instances of present class.
+ * @alpha
+ */
+class RealityDataConnection implements IRealityDataConnection {
+  private _rd: RealityData | undefined;
+  private _rdSource: RealityDataSource;
+
+  private constructor(props: RealityDataSourceProps) {
+    this._rdSource = RealityDataSource.fromProps(props);
+  }
+  /**
+   * Create an instance of this class from a source key and iTwin context/
+   * @alpha
+   */
+  public static async createFromSourceKey(sk: RealityDataSourceKey, iTwinId: GuidString | undefined): Promise<RealityDataConnection | undefined> {
+    const props: RealityDataSourceProps = {sourceKey: sk};
+    const rdConnection = new RealityDataConnection(props);
+    let tilesetUrl: string | undefined;
+    try {
+      await rdConnection.queryRealityData(iTwinId);
+      tilesetUrl = await rdConnection._rdSource.getServiceUrl(iTwinId);
+    } catch (e) {
+    }
+
+    return (tilesetUrl !== undefined) ? rdConnection : undefined;
+  }
+  /**
+   * Query Reality Data from provider
+   */
+  private async queryRealityData(iTwinId: GuidString | undefined) {
+    if (this._rdSource.isContextShare && !this._rd) {
+      const token = await this._rdSource.getAccessToken();
+      if (token && this._rdSource.realityDataId) {
+        const client = new RealityDataClient();      // we need to resolve tilesetURl from realityDataId and iTwinId
+        const requestContext = new AuthorizedFrontendRequestContext(token);
+        this._rd = await client.getRealityData(requestContext,iTwinId, this._rdSource.realityDataId);
+      }
+    }
+  }
+  /**
+   * Returns Reality Data if available
+   */
+  public getRealityData(): RealityData | undefined {
```

Comment:
Replace `getRealityData()` and `getRealityDataType()` with `get realityData` and `get realityDataType` (or just `get type`).